src/UNL/UCBCN/Event.php

```diff
@@ -247,18 +341,23 @@ public function update($do=false)
      *
      * @return int|false
      */
-    public function addToCalendar($calendar_id, $status='pending', $sourcemsg = 'unknown')
+    public function addToCalendar($calendar_id, $status='pending', $sourcemsg = null)
     {
-        $values = array(
-                'calendar_id'     => $calendar_id,
-                'event_id'        => $this->id,
-                'uidcreated'      => $_SESSION['_authsession']['username'],
-                'datecreated'     => date('Y-m-d H:i:s'),
-                'datelastupdated' => date('Y-m-d H:i:s'),
-                'uidlastupdated'  => $_SESSION['_authsession']['username'],
-                'status'          => $status,
-                'source'          => $sourcemsg);
-        return UNL_UCBCN::dbInsert('calendar_has_event', $values);
+        $calendar_has_event = new CalendarHasEvent;
+
+        $calendar_has_event->calendar_id = $this->id;
+        $calendar_has_event->event_id = $this->id;
+        $calendar_has_event->uidcreated = $_SESSION['__SIMPLECAS']['UID'];
+        $calendar_has_event->datecreated = date('Y-m-d H:i:s');
+        $calendar_has_event->datelastupdated = date('Y-m-d H:i:s');
+        $calendar_has_event->uidlastupdated = $_SESSION['__SIMPLECAS']['UID'];
+        $calendar_has_event->status = $status;
+
+        if (isset($source)) {
```

Comment:
The parameter name is `$sourcemsg`, so it doesn't look like `$source` will ever be set here.